flare/otf.py

```diff
@@ -191,69 +202,50 @@ def run(self):
         while self.curr_step < self.number_of_steps:
             # run DFT and train initial model if first step and DFT is on
             if self.curr_step == 0 and self.std_tolerance != 0 and len(self.gp.training_data)==0:
-                # call dft and update positions
-                self.run_dft()
-                dft_frcs = copy.deepcopy(self.structure.forces)
-                new_pos = md.update_positions(self.dt, self.noa,
-                                              self.structure)
+
+                self.initialize_train()
+                new_pos = self.md_step()
                 self.update_temperature(new_pos)
                 self.record_state()
 
-                # make initial gp model and predict forces
-                self.update_gp(self.init_atoms, dft_frcs)
-                if (self.dft_count-1) < self.freeze_hyps:
-                    self.train_gp()
-                    if self.write_model >= 2:
-                        self.gp.write_model(self.output_name+"_model")
-
             # after step 1, try predicting with GP model
             else:
-                self.gp.check_L_alpha()
-                self.pred_func(self.structure, self.gp, self.n_cpus)
+
+                # compute forces and stds with GP
                 self.dft_step = False
-                new_pos = md.update_positions(self.dt, self.noa,
-                                              self.structure)
+                self.compute_properties()
 
                 # get max uncertainty atoms
                 std_in_bound, target_atoms = \
                     is_std_in_bound(self.std_tolerance,
                                     self.gp.hyps[-1], self.structure,
                                     self.max_atoms_added)
 
-                if not std_in_bound:
+                if std_in_bound:
+                    new_pos = self.md_step()
+
+                else:
                     # record GP forces
                     self.update_temperature(new_pos)
```

Comment:
The logic is wrong here; velocities get sent to zero, since new_pos hasn't been updated. (I'll fix this)

Reply:
8544b9f addresses this by mirroring the otf loop on master (new positions are calculated first, and then is_std_in_bound is called)